### PR TITLE
fix(suite): Custom backends are used in all urls

### LINKED ISSUE #8808: Custom backend settings ignored in Details/ Inputs, Outputs section
**Describe the bug**
Custom backend settings ignored in Details/ Inputs, Outputs section.
`Details/Amount` section opens correct custom link.
`Details/Inputs, Outputs` section opens default backend.

**Steps to reproduce:**
1. Set custom backend
2. Open both links in Transaction detail 
3. Compare urls

**Info:**
 - Suite version: desktop 23.6.1 (bddd516378beadfcff94523e7c94834ba14f147f)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.6.1 Chrome/112.0.5615.183 Electron/24.3.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.6.1 Universal (revision c3f6e8f19fd647c73b414322e287ba23cf8a0c7c)
 - Transport: BridgeTransport 2.0.33

**Screenshots:**

Screenshot 2023-06-27 at 9 23 13

opens:https://zec1.trezor.io/tx/72a27c44ae23754276327820d5986a8ee9d1b76a8e05eb1e5d69a3b5a9c465dc

Screenshot 2023-06-27 at 9 23 01

opens:https://staging-zec.trezor.io/tx/72a27c44ae23754276327820d5986a8ee9d1b76a8e05eb1e5d69a3b5a9c465dc

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve #8808

### CODE REVIEW

**Review by tomasklim — CHANGES_REQUESTED** [review]
Please remove `submodules/trezor-common` change

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/IODetails/AnalyzeInExplorerBanner.tsx [hunk 1]
```diff
@@ -54,14 +55,13 @@ const StyledButton = styled(Button)`
...
 }
 
-export const AnalyzeInExplorerBanner = ({ txid }: AnalyzeInExplorerBannerProps) => {
+export const AnalyzeInExplorerBanner = ({ txid, symbol }: AnalyzeInExplorerBannerProps) => {
     const theme = useTheme();
 
-    const { selectedAccount } = useSelector(state => state.wallet);
-    const { network } = selectedAccount;
-    const explorerUrl = network?.explorer.tx;
+    const explorerUrl = useSelector(state => state.wallet.blockchain[symbol].explorer.tx);
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/IODetails/AnalyzeInExplorerBanner.tsx]: use `useExplorerTxUrl` hook

File: packages/suite/src/hooks/suite/useExplorerTxUrl.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
+import { NetworkSymbol } from '@suite-common/wallet-config';
 import { useSelector } from 'src/hooks/suite/useSelector';
 
-export const useExplorerTxUrl = () => {
-    const network = useSelector(state => state.wallet.selectedAccount.network);
+export const useExplorerTxUrl = (symbol: NetworkSymbol) => {
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useExplorerTxUrl.ts]: How about getting rid of this hook and changing it to selector?

**Review by tomasklim — CHANGES_REQUESTED** [review]

**Comment by tomasklim** [comment]
@jvaclavik lets finish this?

**Comment by jvaclavik** [comment]
> @jvaclavik lets finish this?

yes 🫣😅

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/IODetails/AnalyzeInExplorerBanner.tsx [hunk 3]
```diff
@@ -25,19 +27,18 @@ const Description = styled.span`
...
+export const AnalyzeInExplorerBanner = ({ txid, symbol }: AnalyzeInExplorerBannerProps) => {
+    const explorer = useSelector(state => selectBlockchainExplorerBySymbol(state, symbol));
 
     return (
         <NotificationCard
             variant="info"
             icon="CUBE"
             button={{
-                href: `${explorerUrl}${txid}`,
+                href: `${explorer?.address}${txid}`,
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/IODetails/AnalyzeInExplorerBanner.tsx]: Should be `explorer.tx`.

Screenshot 2024-07-16 at 14 45 31

File: packages/suite-desktop/suite-desktop/releaseNotes/release-notes.md
Comment (komret) [inline packages/suite-desktop/suite-desktop/releaseNotes/release-notes.md:1]: Does this belong here?

**Review by tomasklim — APPROVED** [review]
LGTM

**Comment by tomasklim** [comment]
@trezor/qa please test this together with https://github.com/trezor/trezor-suite/pull/13834

**Pull request merged**